### PR TITLE
SSOI integration closes #42

### LINKED ISSUE #42: Integrate SSOI with login check

### PR DESCRIPTION
**Pull request by orischwartz**
For #42

### CODE REVIEW

**Comment by orischwartz** [comment]
@shanear if you want to get a jump on reviewing this ...

####Changes included
* Adds omniauth (via [saml-va](https://github.com/department-of-veterans-affairs/omniauth-saml-va))
* Removes fake ssoi attributes in favor of OmnitAuth's dummy [Developer Strategy](http://www.rubydoc.info/github/intridea/omniauth/master/OmniAuth/Strategies/Developer)
* Customizes developer strategy so to use just one field--`email`, which is used as the authentication's `uid`
* Changes user `*authorized?` methods to reflect real-world usage:
    - User is first prompted to authenticate via SSOI
    - If successful, user is then prompted to authenticate via VACOLS
    - Possible states: unauthenticated, SSOI only, both SSOI and VACOLS

**Comment by shanear** [comment]
on it.

File: lib/fakes/authentication_service.rb [hunk 1]
```diff
@@ -1,11 +1,7 @@
 class Fakes::AuthenticationService
-  cattr_accessor :vacols_regional_offices, :ssoi_username, :ssoi_enabled
+  cattr_accessor :vacols_regional_offices
 
   def self.authenticate_vacols(regional_office, password)
     vacols_regional_offices[regional_office] == password
   end
-
-  def self.ssoi_authentication_enabled?
```
Comment (shanear) [inline lib/fakes/authentication_service.rb]: :heart:

**Comment by shanear** [comment]
this looks awesome

waiting on travis :turtle:

**Comment by paultag** [comment]
Awesome! Was just able to login via SSOi!

image

**Pull request merged**